nativelib/src/main/resources/posix/netdb.c

```diff
@@ -0,0 +1,86 @@
+#include "netdb.h"
+#include "sys/socket_conversions.h"
+#include <stddef.h>
+#include <stdlib.h>
+
+void scalanative_convert_scalanative_addrinfo(struct scalanative_addrinfo *in,
+				  struct addrinfo *out) {
+	out->ai_flags = in->ai_flags;
+	out->ai_family = in->ai_family;
+	out->ai_socktype = in->ai_socktype;
+	out->ai_protocol = in->ai_protocol;
+	out->ai_canonname = in->ai_canonname;
+	if(in->ai_addr == NULL) {
+		out->ai_addr = NULL;
+		out->ai_addrlen = in->ai_addrlen;
+	} else {
+		struct sockaddr *converted_addr = malloc(in->ai_addrlen);
+		socklen_t *len = malloc(sizeof(socklen_t));
```

Comment:
Yeah, I've got to free `len`. `converted_addr` though is a part of struct returned from the function, so I guess it should be freed in `freeaddrinfo`?